engine/pkg/client/errors.go

```diff
@@ -22,7 +22,8 @@ import (
 type ExecutorNotFoundErrInfo struct {
 	rpcerror.Error[rpcerror.Retryable, rpcerror.NotFound]
 
-	ExecutorID model.ExecutorID
+	ExecutorID  model.ExecutorID
```

Comment:
If we have limited online executors, why don't we check the `online` executor list? so we don't need to do some gc.

Reply:
Anything by designed here?

Reply:
We need a list of "executors definitely not online" in addition to a list of "executors currently online". For example, if you want a client for `executor-1`, it is possible that

1. The executor has gone offline, or
2. The executor has just gotten online, but due to a consistency problem, we just don't have a record for it yet.

It is worthwhile distinguishing these two situations because in situation 1, waiting for a while will be futile, just leading to wasted latency.